server/controllers/wip/questionsController.ts

```diff
@@ -92,6 +97,85 @@ export default class QuestionsController extends BaseController<FormWizard.Multi
     })
   }
 
+  async saveValues(
+    req: FormWizard.Request<FormWizard.MultiValues, string>,
+    res: express.Response,
+    next: express.NextFunction,
+  ): Promise<void> {
+    const { incidentReportingApi } = res.locals.apis
+
+    const submittedValues = req.form.values
+
+    const report = res.locals.report as ReportWithDetails
+    const reportConfig = res.locals.reportConfig as IncidentTypeConfiguration
+
+    const questionsResponses = []
+    for (const [fieldName, values] of Object.entries(submittedValues)) {
+      const questionConfig = reportConfig.questions[fieldName]
+      if (questionConfig === undefined) {
+        logger.error(
+          new Error(
+            `Report '${report.id}': Submitted Field '${fieldName}' not found in ${report.type}'s configuration.`,
+          ),
+        )
+
+        // eslint-disable-next-line no-continue
+        continue
+      }
+      const questionResponses: AddOrUpdateQuestionWithResponsesRequest = {
+        code: fieldName,
+        question: questionConfig.code,
+        additionalInformation: null,
+        responses: [],
+      }
+
+      const responseCodes = (questionConfig.multipleAnswers ? values : [values]) as string[]
+      for (const responseCode of responseCodes) {
+        if (responseCode === '') {
+          // eslint-disable-next-line no-continue
+          continue
+        }
+        const answerConfig = this.findAnswerConfigByCode(responseCode, questionConfig)
+        if (answerConfig === undefined) {
+          logger.error(
+            new Error(
+              `Report '${report.id}': Submitted Answer with code '${responseCode}' not found in ${report.type}'s question '${questionConfig.id}' configuration.`,
+            ),
+          )
+          // eslint-disable-next-line no-continue
+          continue
+        }
+
+        const response: AddOrUpdateQuestionResponseRequest = {
+          response: responseCode,
+          responseDate: null,
+          additionalInformation: null,
+        }
+
+        if (answerConfig.commentRequired) {
+          const commentFieldName = `${questionConfig.id}-${answerConfig.id}-comment`
+          if (submittedValues[commentFieldName] !== '') {
```

Comment:
No. This is because the submitted values include all fields for the step, including all conditional (comment/date) fields for these steps, even when the user didn't select the relative answer.

For example

Question:
WHAT IS THE REASON FOR THIS FOOD REFUSAL?
- [X] LOCATION
- [ ] OTHER (PLEASE SPECIFY)
  - value for this field not required but submitted as '' by FormWizard

Validation works and it happens before reaching the saving hook but the values for these submitted fields don't need to be submitted to the API.

(if I didn't skip the subsequent bit where it reads the configuration for the question for `${question.id}-${answer.id}-comment` would not find the question anyway as this is not the code for a question)

Reply:
And a empty string couldn't be matched against the possible responses from the report configuration anyway (the response code, e.g. `NECK` is used to find the corresponding answer config)

Reply:
> even when the user didn't select the relative answer
eww. i guess it's to be expected: the conditionally-revealed box shouldn't _force_ disable all contained input elements!

Reply:
actually: what if a user selects radio button 1, types a comment "A", changes their mind, selects radio button 2, types a comment "B"… what is inside form values for the comment? is it an array of both, just the first or just the last???

Reply:
> actually: what if a user selects radio button 1, types a comment "A", changes their mind, selects radio button 2, types a comment "B"… what is inside form values for the comment? is it an array of both, just the first or just the last???

I can't remember if I've seen this but I think only the actual value for the selected one would go through. Then as we construct the API request from questions/selected answers anything else would be ignored.

Reply:
As discussed offline, I thought you were talking about the other `if` statement at the top of the logic.
These if to check that the date/comment values are non-empty strings should not be necessary: If the request reached this point the data is already validated: [Removed unnecessary checks on validated data](https://github.com/ministryofjustice/hmpps-incident-reporting/pull/281/commits/0584ad17dac9607fa97236e72bcc1db151c029f1)